stable_baselines3/common/buffers.py

```diff
@@ -424,7 +424,7 @@ def compute_returns_and_advantage(self, last_values: th.Tensor, dones: np.ndarra
         last_gae_lam = 0
         for step in reversed(range(self.buffer_size)):
             if step == self.buffer_size - 1:
-                next_non_terminal = 1.0 - dones
+                next_non_terminal = 1.0 - dones.astype(int)
```

Comment:
```suggestion
                next_non_terminal = 1.0 - dones.astype(np.float32)
```

to be consistent with `next_non_terminal = 1.0 - self.episode_starts[step + 1]`, this should still fix the issue, no?